metrics/circonus.go

```diff
@@ -0,0 +1,132 @@
+package metrics
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	cgm "github.com/circonus-labs/circonus-gometrics"
+)
+
+type cgmProvider struct {
+	metrics *cgm.CirconusMetrics
+	prefix  string
+}
+
+type cgmTimer struct {
+	metrics *cgm.CirconusMetrics
+	name    string
+}
+
+var (
+	circonus *cgmProvider
+	once     sync.Once
+)
+
+const serviceName = "fabio"
+
+// circonusBackend returns a provider that reports to Circonus.
+func circonusBackend(prefix string,
+	circKey string,
+	circApp string,
+	circURL string,
+	circBrokerID string,
+	circCheckID string,
+	interval time.Duration) (Registry, error) {
+
+	var initError error
+
+	once.Do(func() {
+
+		if circKey == "" {
+			initError = errors.New("metrics: Circonus API token key")
+			return
+		}
+
+		if circApp == "" {
+			circApp = serviceName
+		}
+
+		host, err := os.Hostname()
+		if err != nil {
+			initError = fmt.Errorf("metrics: unable to initialize Circonus %s", err)
+			return
+		}
+
+		cfg := &cgm.Config{}
+
+		cfg.CheckManager.API.TokenKey = circKey
+		cfg.CheckManager.API.TokenApp = circApp
+		cfg.CheckManager.API.URL = circURL
+		cfg.CheckManager.Check.ID = circCheckID
+		cfg.CheckManager.Broker.ID = circBrokerID
+		cfg.Interval = fmt.Sprintf("%.0fs", interval.Seconds())
+		cfg.CheckManager.Check.InstanceID = host
+		cfg.CheckManager.Check.DisplayName = fmt.Sprintf("%s /%s", host, serviceName)
+		cfg.CheckManager.Check.SearchTag = fmt.Sprintf("service:%s", serviceName)
+
+		metrics, err := cgm.NewCirconusMetrics(cfg)
+		if err != nil {
+			initError = fmt.Errorf("metrics: unable to initialize Circonus %s", err)
+		} else {
+			circonus = &cgmProvider{metrics, prefix}
+		}
+
+		metrics.Start()
+
+		log.Print("[INFO] Sending metrics to Circonus")
+
+	})
+
+	return circonus, initError
+}
+
+// Names returns the list of registered metrics acquired
```

Comment:
Pls drop the interface comments. Just leave the comments that explain that circonus does not support this.